# Hi, I'm Fred! ðŸ‘‹

:australia: *Senior Bioinformatician at the [Sydney Informatics Hub](https://github.com/Sydney-Informatics-Hub) and the [Australian BioCommons](https://github.com/AustralianBioCommons)* :australia:

I enjoy wearing many hats - most of which aim to **improve the accessibility of computational and data-heavy research** for scientists, and to **develop technical resources that help translate bioinformatics findings to actionable insights** towards epidemiology and conservation purposes.

Always keen to chat, and discuss open-source projects - feel free to **get in touch**, and **take a peek at a few repos** below!

:mortar_board:: [Google Scholar](https://scholar.google.com.au/citations?user=LxKibdoAAAAJ&hl=en)  
:email:: `frederick[dot]jaya[at]sydney.edu.au`

## Software :technologist:

Significant contributions to scientific libraries, packages, and tools during my time at the Australian National University. Mainly Python and R. 

- :dna: **[cogent3](https://github.com/cogent3/cogent3)**: A python library for analysis of genomic sequence data with a flexible plugin architecture to interface with other tools. 
- **[PoolTools](https://github.com/AngusMcLure/PoolTools)**: An easy-to-use Shiny interface to R packages for the design and analysis of pooled data.
- **[PoolPower](https://github.com/AngusMcLure/PoolPoweR)**: An R package for power and sample size calculations for designing optimal pool testing surveys.

## Workflows :mechanic:

Development of scalable Nextflow pipelines, most are optimised for use on institutional or national HPCs. Some fortunate enough to have led to publications! 

- :test_tube: **[rec-bench](https://github.com/fredjaya/rec-bench)**: Nextflow pipeline for benchmarking tools that detect recombination in virus sequences.
  - :newspaper:	*[Jaya et al. (2024); Virus Evol.](https://academic.oup.com/ve/article/9/2/vead066/7444193?login=false)*
- :koala: **[phasco-genes](https://github.com/fredjaya/phasco_genes)**: Pre-processing pipeline for finding genome-wide associations to environmental variables.
  - ðŸ“° *[McLennan et al. (2024); Ecol. Appl.](https://doi.org/10.1002/eap.3062)*
- :microbe: **[ONT-bacpac-nf](https://github.com/Sydney-Informatics-Hub/ONT-bacpac-nf)**: Nextflow pipeline to assemble bacterial genomes from Nanopore long read data and identify AMR genes. 

## Teaching :teacher:

Past workshop content to help lower the barrier to bioinformatics.

- :twisted_rightwards_arrows:	**[hello-nextflow](https://github.com/Sydney-Informatics-Hub/hello-nextflow)**: A hands-on introduction to Nextflow. Written for biologists, by biologists. Part of the National Bioinformatics Training Cooperative.
- :palm_tree: **[spw23_iqtree](https://github.com/fredjaya/spw23_iqtree)**: What is IQ-TREE and how do you use it? Part of the Sydney Phylogenetics Workshop.

## Hobby :artist:

Non-bioinformatics miscellanea when a big coding/technical itch needs to be scratched.

- :keyboard: **[zmk-config](https://github.com/fredjaya/zmk-config)**: Key mappings to accompany custom mechanical keyboards.
- :page_facing_up: **[dotfiles](https://github.com/fredjaya/dotfiles)**: Lazy ass bash aliases and configs for F1 CLI-ing.
- :space_invader: **[chickenpy](https://github.com/fredjaya/chickenpy)**: Cooking is modular. OOP is modular. Why not try combine the two? Might be usable one day.
